layouts/_default/single.html

```diff
@@ -2,9 +2,11 @@
   <div class="post">
     <h2 class="post-title"><a href="{{ .Permalink }}">{{ .Title | markdownify }}</a></h2>
     <div class="post-meta">
-      <span class="post-date">
-        {{ .Date.Format "2006-01-02" }}
-      </span>
+      {{ with .Date | default "" }}
```

Comment:
Is it necessary to set `default` here? 🤔

Reply:
Sadly it is (I am also not that happy with it), the problem is that an empty date defaults to `0001-01-01` and an `if` or `with` statement seems to convert it to this string when checked. If you have a better solution: I'm happy to see it :)

Reply:
Huh, I thought that `if` statement doesn't require any context or default value. I'll check it later ;-)

Reply:
I changed it to use `nil` (seems more meaningful) and nope, an `if` has the same problem